msteams-platform/messaging-extensions/how-to/search-commands/respond-to-search.md

```diff
@@ -80,13 +80,127 @@ Your service must respond with the results matching the user query. The response
+    var settings = urlParams.get('settings');
+    if (settings) 
+    {
+        var checkboxes = document.getElementsByName("includeInSearch");
+        for (var i = 0; i < checkboxes.length; i++) 
+        {
+            var thisCheckbox = checkboxes[i];
+            if (settings.includes(thisCheckbox.value)) 
+            {
+                checkboxes[i].checked = true;
+            }
+        }
+    }
+});
+</script>
+<script type="text/javascript">
+microsoftTeams.initialize();
+function onSubmit() 
+{
+    var newSettings = [];
+    var checkboxes = document.getElementsByName("includeInSearch");
+    for (var i = 0; i < checkboxes.length; i++) 
+    {
+        if (checkboxes[i].checked) 
+        {
+            newSettings.push(checkboxes[i].value);
+        }
+    }
+    microsoftTeams.authentication.notifySuccess(JSON.stringify(newSettings));
+}
+</script>
+</body>
+</html>
+```
+
+The response from the configuration page is also handled by onQuery. The sole exception is when the configuration page is called by the handler for onQuerySettingsUrl. The handler for onQuerySettingsUrl returns the URL for the configuration page. After the configuration page closes, the handler for onSettingsUpdate accepts and saves the returned state as shown in the following example:
+
+```csharp
+
+protected override async Task<MessagingExtensionResponse> OnTeamsMessagingExtensionConfigurationQuerySettingUrlAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+{
+    // The user has requested the Messaging Extension Configuration page.
+    var escapedSettings = string.Empty;
+    var userConfigSettings = await _userConfigProperty.GetAsync(turnContext, () => string.Empty);
+    if (!string.IsNullOrEmpty(userConfigSettings))
+    {
+        escapedSettings = Uri.EscapeDataString(userConfigSettings);
+    }
+    return new MessagingExtensionResponse
+    {
+        ComposeExtension = new MessagingExtensionResult
+            {
+                Type = "config",
+                SuggestedActions = new MessagingExtensionSuggestedAction
+                    {
+                        Actions = new List<CardAction>
+                            {
+                                new CardAction
+                                {
+                                    Type = ActionTypes.OpenUrl,
+                                    Value = $"{_siteUrl}/searchSettings.html?settings={escapedSettings}",
+                                },
+                            },
+                    },
+            },
+    };
+}
+
+```
+
+### `auth` response type
+
+If your service requires user authentication, the users must sign in before they use the message extension. For more information, see [authentication](~/messaging-extensions/how-to/add-authentication.md#authentication).
+
+### `message` response type
+
+`message` is used when your extension needs to display a plain text message.
```

Comment:
What does this look like?

Reply:
Does it support formatting? We need to be explicit-- this is a frequently asked question.